corehq/apps/data_dictionary/management/commands/refresh_data_dictionary.py

```diff
@@ -0,0 +1,81 @@
+from collections import defaultdict
+
+from django.core.management.base import BaseCommand
+from django.http import Http404
+
+from corehq.apps.app_manager.const import USERCASE_TYPE
+from corehq.apps.app_manager.dbaccessors import get_app, get_apps_in_domain
+from corehq.apps.app_manager.exceptions import AppInDifferentDomainException
+from corehq.apps.data_cleaning.utils.cases import (
+    clear_caches_case_data_cleaning,
+)
+from corehq.apps.domain.models import Domain
+from corehq.apps.domain_migration_flags.api import (
+    get_migration_complete,
+    set_migration_complete,
+    set_migration_started,
+)
+from corehq.toggles import VELLUM_SAVE_TO_CASE
+from corehq.util.log import with_progress_bar
+
+
+class Command(BaseCommand):
+    help = 'Refreshes data dictionary for all domains and their apps. ' \
+           'For specific domains, ./manage.py refresh_data_dictionary domain1 domain2'
+
+    def add_arguments(self, parser):
+        parser.add_argument('domains', nargs='*',
+            help="Domain name(s). If blank, will refresh for all domains")
+
+    def handle(self, **options):
+        domains = options['domains'] or Domain.get_all_names()
+
+        for domain in with_progress_bar(domains):
+            if get_migration_complete(domain, 'refresh_data_dictionary'):
+                continue
+            set_migration_started(domain, 'refresh_data_dictionary')
+            try:
+                apps = get_apps_in_domain(domain)
+                for app in apps:
+                    try:
+                        _refresh_data_dictionary_from_app(domain, app.get_id)
+                    except Exception as e:
+                        print(f'Failed to refresh app {app.get_id} in domain {domain}: {str(e)}')
+                clear_caches_case_data_cleaning(domain)
+
+            except Exception as e:
+                print(f'Failed to get apps in domain {domain}: {str(e)}')
+            set_migration_complete(domain, 'refresh_data_dictionary')
+
+
+def _refresh_data_dictionary_from_app(domain, app_id):
+    # This is an exact copy of the function in corehq.apps.app_manager.tasks
```

Comment:
why don't we share the code as a separate utility that's then called from the task and here, rather than duplicating the code? seems like we could miss that this duplicated code exists here and it feels un-DRY

Reply:
some folks are comment-blind, so they won't take the time to read docstrings/comments and understand that this is a duplication. also increases maintenance burden

Reply:
My expectation is this command should be only running once, then the original `refresh_data_dictionary_from_app` function will take care of future refresh. So I didn't expect anyone to use and read this command except for me. But I guess no harm to create a separate utility in the tasks file.